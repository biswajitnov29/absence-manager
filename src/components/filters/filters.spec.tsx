import React from "react";
import { render, screen } from "@testing-library/react";
import SiteFilters from "./filters";

const statusListTest:string[] = ['Requested', 'Confirmed', 'Rejected'];
describe('<SiteFilters />', () => {
    test('renders correctly', () => {
        const { container } = render(<SiteFilters statusList={statusListTest} />);
        expect(container).toMatchSnapshot();
    });
    test('check status in the list', () => {
        render(<SiteFilters statusList={statusListTest} />);
        expect(screen.getByText('Requested')).toBeDefined();
    });
});