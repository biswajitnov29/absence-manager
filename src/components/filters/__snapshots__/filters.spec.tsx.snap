// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteFilters /> renders correctly 1`] = `
<div>
  <div
    class="site-box"
  >
    <div
      class="site-field"
    >
      <select
        class="site-field-item"
        id="filter-status"
      >
        <option
          value=""
        >
          Select Status
        </option>
        <option
          value="Requested"
        >
          Requested
        </option>
        <option
          value="Confirmed"
        >
          Confirmed
        </option>
        <option
          value="Rejected"
        >
          Rejected
        </option>
      </select>
      <label
        class="site-field-select-label"
        for="filter-status"
      >
        Filter By Status
      </label>
    </div>
    <div
      class="site-field"
    >
      <input
        class="site-field-item"
        id="filter-daterange"
        name="birthday"
        type="date"
      />
      <label
        class="site-field-select-label"
        for="ilter-daterange"
      >
        Filter By Date
      </label>
    </div>
  </div>
</div>
`;
